### ISSUE TITLE
Anchor create / detach can block the update thread for significant periods

### ISSUE THREAD

**Issue by tangobravo** [google-ar/arcore-android-sdk#1722]
### SPECIFIC ISSUE ENCOUNTERED

Calling `createAnchor` or `anchor.detach()` sometimes blocks for a significant amount of time (hundreds of ms).

From a perfetto trace it looks like it is blocked until the MTC_mapping thread is idle. That thread looks like it has a bigger workload every few seconds, and can sometimes be hundreds of ms, especially if the camera has been moving rapidly / exploring new areas in the previous period.

### VERSIONS USED

- Android Studio: Android Studio Meerkat | 2024.3.1
- ARCore SDK for Android: v1.48.0
- Device manufacturer, model, and O/S: Samsung Galaxy S23, Android 14
- Google Play Services for AR (ARCore):  1.48.250340293
- Output of `adb shell getprop ro.build.fingerprint`: samsung/dm1qxeea/dm1q:14/UP1A.231005.007/S911BXXS8CYBD:user/release-keys

### STEPS TO REPRODUCE THE ISSUE
I've modified hello_ar_java from the samples to make this easier to reproduce.

You can find my modified version here: https://github.com/tangobravo/arcore-android-sdk

The changes mean that anchors will constantly be created and detached as long as a finger is held down, which makes this easier to reproduce given that it only reproduces if the anchor update is requested whilst the mapping thread is busy.

Once a plane is detected, hold your finger on it and sweep the phone around relatively fast. If the createAnchor call takes more than 10ms, it prints a warning in logcat.

### WORKAROUNDS (IF ANY)

I can't find any user-visible state to know beforehand if the anchor function is going to block for a significant period, so there's no way I can see to avoid the jank, other than avoiding placing anchors at all.

### ADDITIONAL COMMENTS

Perfetto data that indicates there's some sync point here with the MTC_mapping thread that can sometimes take significant time:
<img width="613" alt="Image" src="https://github.com/user-attachments/assets/91b1bd0b-55ff-436e-b451-74934a0579ab" />

Snippet of logcat around a slower createAnchor call:
```
2025-04-25 13:29:32.749  9552-9585  HelloArActivity         com...ar.core.examples.java.helloar  I  Create anchor took 0.1 ms
2025-04-25 13:29:32.749  9552-9585  native                  com...ar.core.examples.java.helloar  I  I0000 00:00:1745584172.749220    9585 cloud_anchor_manager.cc:530] Anchor Hosting is not configured.
2025-04-25 13:29:32.784  9552-9585  qdgralloc               com...ar.core.examples.java.helloar  W  getInterlacedFlag: getMetaData returned 3, defaulting to interlaced_flag = 0
2025-04-25 13:29:32.785  9552-9585  native                  com...ar.core.examples.java.helloar  E  E0000 00:00:1745584172.785413    9585 hit_test.cc:426] INTERNAL: No point hit.
2025-04-25 13:29:32.786  9552-9585  HelloArActivity         com...ar.core.examples.java.helloar  I  Create anchor took 0.0 ms
2025-04-25 13:29:32.786  9552-9585  native                  com...ar.core.examples.java.helloar  I  I0000 00:00:1745584172.786402    9585 cloud_anchor_manager.cc:530] Anchor Hosting is not configured.
2025-04-25 13:29:32.801  9552-9644  native                  com...ar.core.examples.java.helloar  I  I0000 00:00:1745584172.801506    9644 viwls_helper.cc:240] [M] LiteCOM fixed 40/15.3% points and 0/0.0% keyframes.
2025-04-25 13:29:32.810  9552-9585  qdgralloc               com...ar.core.examples.java.helloar  W  getInterlacedFlag: getMetaData returned 3, defaulting to interlaced_flag = 0
2025-04-25 13:29:32.811  9552-9585  native                  com...ar.core.examples.java.helloar  E  E0000 00:00:1745584172.811086    9585 hit_test.cc:426] INTERNAL: No point hit.
2025-04-25 13:29:32.873  9552-9644  native                  com...ar.core.examples.java.helloar  I  I0000 00:00:1745584172.873445    9644 viwls_optimization.cc:1345] [M] MAP SOLVE: USER_SUCCESS
2025-04-25 13:29:32.889  9552-9585  HelloArActivity         com...ar.core.examples.java.helloar  W  Create anchor took 78.2 ms
2025-04-25 13:29:32.889  9552-9585  native                  com...ar.core.examples.java.helloar  I  I0000 00:00:1745584172.889719    9585 cloud_anchor_manager.cc:530] Anchor Hosting is not configured.
2025-04-25 13:29:32.892  9552-9585  qdgralloc               com...ar.core.examples.java.helloar  W  getInterlacedFlag: getMetaData returned 3, defaulting to interlaced_flag = 0
2025-04-25 13:29:32.921  9552-9585  native                  com...ar.core.examples.java.helloar  E  E0000 00:00:1745584172.921862    9585 hit_test.cc:426] INTERNAL: No point hit.
2025-04-25 13:29:32.922  9552-9585  HelloArActivity         com...ar.core.examples.java.helloar  I  Create anchor took 0.0 ms
2025-04-25 13:29:32.922  9552-9585  native                  com...ar.core.examples.java.helloar  I  I0000 00:00:1745584172.922720    9585 cloud_anchor_manager.cc:530] Anchor Hosting is not configured.
2025-04-25 13:29:32.928  9552-9585  qdgralloc               com...ar.core.examples.java.helloar  W  getInterlacedFlag: getMetaData returned 3, defaulting to interlaced_flag = 0
2025-04-25 13:29:32.931  9552-9585  native                  com...ar.core.examples.java.helloar  E  E0000 00:00:1745584172.931227    9585 hit_test.cc:426] INTERNAL: No point hit.
2025-04-25 13:29:32.931  9552-9585  HelloArActivity         com...ar.core.examples.java.helloar  I  Create anchor took 0.0 ms
2025-04-25 13:29:32.931  9552-9585  native                  com...ar.core.examples.java.helloar  I  I0000 00:00:1745584172.931850    9585 cloud_anchor_manager.cc:530] Anchor Hosting is not configured.
```

This seems to reproduce across devices. I've seen pauses of over a second reported on Pixel 8A for example.

**Comment by tangobravo** [comment]
Just to note it does also occur with individual short taps, if you get unlucky with the timing:
<img width="1084" alt="Image" src="https://github.com/user-attachments/assets/785793e9-4c9a-4938-8827-6e86f88baefc" />

So it doesn't seem to be caused by having too many anchor changes in a map update window. It is significantly easier to reproduce by holding down your finger with the test app though.

**Comment by RGregat** [comment]
I’ve noticed something similar. My application creates and removes anchors based on external events. Sometimes multiple anchors are removed at once, sometimes only a single one. So far, I haven’t found any pattern related to the number of anchors I create or detach that would reproduce the hiccup — it just happens, and it **can** get worse over time, sometimes leading to delays of several seconds, which usually results in a crash.

**Comment by RGregat** [comment]
Interesting, what I have noticed is, that the deletion of an Anchor will more likely result in a freeze than the creation. The workaround to use a World-Anchor is interesting, but I think not adaptable to my usecase. 
A little context what I'm doing at the moment. I combine GNSS with ARCore and one part of it, is a live transformation based on the camera pose and the gnss antenna. It works, but it could be more accurate if I could anchor certain camera pose. we talk about a handful max.

**Comment by agGitHub** [comment]
Same issue here. Any solution ?
